.github/workflows/sync-translations-poeditor.yml

```diff
@@ -0,0 +1,185 @@
+name: Sync Translations with POEditor
+
+on:
+  push:
+    branches:
+      - feat/translation-poeditor-integration
+    paths:
+      - "src/assets/locale/**"
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  sync-translations:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up node
+        uses: actions/setup-node@v4
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Configure Git
+        run: |
+          git config user.name ${{ secrets.GIT_BOT_USERNAME }}
+          git config user.email ${{ secrets.GIT_BOT_EMAIL }}
+          git remote set-url origin https://${{ secrets.GIT_BOT_ACCESS_TOKEN }}@github.com/Aam-Digital/ndb-core.git
+
+      - name: Create or update branch for translations
+        run: |
+          branch_name="update-translations-messages"
+
+          # Check if branch exists on remote
+          if git ls-remote --exit-code origin $branch_name; then
+            echo "Branch $branch_name already exists. Fetching it..."
+            git fetch origin $branch_name:$branch_name
+
+            git checkout $branch_name
+            git pull origin $branch_name
+          else
+            echo "Creating new branch $branch_name..."
+            git checkout -b $branch_name
+          fi
+
+      - name: Extract i18n keys
+        run: npm run extract-i18n
+
+      - name: Upload translations to POEditor
+        run: |
+          declare -A files
+
+          # Populate file mappings dynamically
+          files["en"]="src/assets/locale/messages.xlf"
+          files["de"]="src/assets/locale/messages.de.xlf"
+          files["it"]="src/assets/locale/messages.it.xlf"
+          files["fr"]="src/assets/locale/messages.fr.xlf"
+
+          for lang in "${!files[@]}"; do
+            echo "Uploading ${files[$lang]} for language $lang"
+            response=$(curl -s -X POST https://api.poeditor.com/v2/projects/upload \
+              -F "api_token=${{ secrets.POEDITOR_API_TOKEN }}" \
+              -F "id=${{ secrets.POEDITOR_PROJECT_ID }}" \
+              -F "language=$lang" \
+              -F "updating=terms_translations" \
+              -F "overwrite=1" \
+              -F "sync_terms=1" \
+              -F "file=@${files[$lang]}")
+            status=$(echo $response | jq -r '.response.status')
+            if [ "$status" != "success" ]; then
+              echo "Error uploading file for $lang: $response"
+              exit 1
+            fi
+            echo "Waiting for 30 second to respect rate limits..."
+            sleep 30
```

Comment:
Added the sleep interval, ensuring compliance with the rate limit of one request every 20 seconds.